bson/src/main/org/bson/types/Decimal128.java

```diff
@@ -374,6 +373,84 @@ public boolean isNaN() {
         return (high & NaN_MASK) == NaN_MASK;
     }
 
+
+    /**
+     * Converts this {@code Decimal128} to a {@code int}. This conversion is analogous to the <i>narrowing primitive conversion</i> from
+     * {@code double} to {@code int} as defined in <cite>The Java&trade; Language Specification</cite>: any fractional part of this
+     * {@code Decimal128} will be discarded, and if the resulting integral value is too big to fit in a {@code int}, only the
+     * low-order 32 bits are returned. Note that this conversion can lose information about the overall magnitude and precision of this
+     * {@code Decimal128} value as well as return a result with the opposite sign. Note that {@code #NEGATIVE_ZERO} is converted to
+     * {@code 0}.
+     *
+     * @return this {@code Decimal128} converted to a {@code int}.
+     * @since 3.10
+     */
+    @Override
+    public int intValue() {
+        return Double.valueOf(doubleValue()).intValue();
```

Comment:
Out of interest, why box the primitive for it just to be cast by the Double class?  Essentially this is boils down to:

 `return (int) doubleValue();`

Reply:
I was just trying to rely on  the correctness to the Double class. But it looks cleaner with the cast so I changed it as you suggest